core/runtime/src/main/java/io/quarkus/runtime/init/InitializationTaskRecorder.java

```diff
@@ -0,0 +1,29 @@
+package io.quarkus.runtime.init;
+
+import java.util.stream.StreamSupport;
+
+import org.eclipse.microprofile.config.ConfigProvider;
+
+import io.quarkus.runtime.PreventFurtherStepsException;
+import io.quarkus.runtime.annotations.Recorder;
+
+/**
+ * A {@link Recorder} that is used to check if the application should exit once all initialization tasks are completed.
+ */
+@Recorder
+public class InitializationTaskRecorder {
+
+    private static final String QUARKUS_INIT_AND_EXIT = "quarkus.initAndExit";
```

Comment:
Minor complaint: This should probably be `quarkus.init-and-exit` if we are to stick with conventions